tests/src/org.hibernate.orm/hibernate-core/7.0.0.Final/src/test/java/org_hibernate_orm/hibernate_core/OrmAnnotationHelperTest.java

```diff
@@ -0,0 +1,147 @@
+            }
+        });
+    }
+
+    private static Map<String, Object> createAttributeMap(AnnotationDescriptor<?> annotationDescriptor) {
+
+        Map<String, Object> attributes = new LinkedHashMap<>(annotationDescriptor.getAttributes().size());
+
+        for (AttributeDescriptor<?> attributeDescriptor : annotationDescriptor.getAttributes()) {
+
+            Class<?> valueType = attributeDescriptor.getTypeDescriptor().getValueType();
+
+            if (valueType.isArray()) {
+
+                ValueTypeDescriptor<?> typeDescriptor = attributeDescriptor.getTypeDescriptor();
+                attributes.put(attributeDescriptor.getName(),
+                    Array.newInstance(typeDescriptor.getValueType().getComponentType(), 0));
+            } else {
+                if (valueType == int.class || valueType == Integer.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == long.class || valueType == Long.class) {
+                    attributes.put(attributeDescriptor.getName(), 0L);
+                } else if (valueType == boolean.class || valueType == Boolean.class) {
+                    attributes.put(attributeDescriptor.getName(), false);
+                } else if (valueType == short.class || valueType == Short.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == byte.class || valueType == Byte.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == char.class || valueType == Character.class) {
+                    attributes.put(attributeDescriptor.getName(), 'c');
+                } else if (valueType == float.class || valueType == Float.class) {
+                    attributes.put(attributeDescriptor.getName(), 0f);
+                } else if (valueType == double.class || valueType == Double.class) {
+                    attributes.put(attributeDescriptor.getName(), 0d);
+                }
+            }
+        }
+        return attributes;
+    }
+
+    private static void createInstances(AnnotationDescriptor<?> annotationDescriptor, AttributeDescriptor<?> attributeDescriptor, Map<String, Object> attributes, BasicModelsContextImpl ctx) {
+
+        for (Annotation usage : usages(ctx, annotationDescriptor, attributes)) {
+            instanceViaUsage(attributeDescriptor, ctx, usage);
+        }
+
+        instanceViaCtors(annotationDescriptor);
+    }
+
+    private static void instanceViaCtors(AnnotationDescriptor<?> annotationDescriptor) {
+        if (annotationDescriptor instanceof MutableAnnotationDescriptor<?, ?> mad) {
+
+            try {
+                Constructor<?> constructor = mad.getMutableAnnotationType().getConstructor(annotationDescriptor.getAnnotationType(), ModelsContext.class);
+                constructor.newInstance(null, null);
+            } catch (Exception e) {
```

Comment:
Be more specific.

Which kind of exceptions are we willing to ignore here? Any or just `ReflectiveOperationException`s?

Reply:
Any, IIRC the exceptions throw did not follow a consistent pattern.

Reply:
Now with [this](https://github.com/oracle/graalvm-reachability-metadata/pull/579#discussion_r2175103105) comment, I can see why `catch (Exception e)` is not a problem here.